src/components/tools/geometry-tool/geometry-toolbar.tsx

```diff
@@ -0,0 +1,124 @@
+// tslint:disable:jsx-no-lambda
+import * as React from "react";
+import { BaseComponent } from "../../base";
+import { IBaseProps } from "../../base";
+import { observer, inject } from "mobx-react";
+import { ToolTileModelType } from "../../../models/tools/tool-tile";
+
+interface IProps extends IBaseProps {
+  model: ToolTileModelType;
+  onAngleLabelClick: () => void;
+  angleLabelDisabled: boolean;
+  angleLabelSelected: boolean;
+  onDeleteClick: () => void;
+  deleteDisabled: boolean;
+  onDuplicateClick: () => void;
+  duplicateDisabled: boolean;
+}
+interface IState {
+  showSettings: boolean;
+}
+
+@inject("stores")
+@observer
+export class GeometryToolbarView extends BaseComponent<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      showSettings: false,
+    };
+  }
+  public render() {
+    return (
+      <div className="geometry-toolbar" data-test="geometry-toolbar" onMouseDown={this.handleMouseDown}>
+        <div className="toolbar-buttons">
+          {this.renderToolHeader(false)}
+          {this.renderToolButton("Select", "selection", null,
+                                  false, false, true)}
```

Comment:
Multiple boolean parameters like this are hard for the reader of the code to interpret. I'd be inclined to define the function something like this:
```
interface IRenderToolButtonParams {
  onClick: (e: MouseEvent) => void;
  selected: boolean;
  disabled: boolean;
  hidden: boolean;
}

private renderToolButton = (toolName: string, toolClass: string, params: IRenderToolButtonParams) => {
}

Reply:
Yup.  Much easier to parse. Will update.